test/hotspot/jtreg/runtime/os/HugePageConfiguration.java

```diff
@@ -0,0 +1,222 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+
+import java.io.*;
+import java.util.Set;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class HugePageConfiguration {
+
+    Set<Long> _staticHugePageSizes;
+    long _staticDefaultHugePageSize;
+
+    enum THPMode {always, never, madvise}
+    THPMode _thpMode;
+    long _thpPageSize;
+
+    public Set<Long> getStaticHugePageSizes() {
+        return _staticHugePageSizes;
+    }
+
+    public long getStaticDefaultHugePageSize() {
+        return _staticDefaultHugePageSize;
+    }
+
+    public THPMode getThpMode() {
+        return _thpMode;
+    }
+
+    public long getThpPageSize() {
+        return _thpPageSize;
+    }
+
+    public HugePageConfiguration(Set<Long> _staticHugePageSizes, long _staticDefaultHugePageSize, THPMode _thpMode, long _thpPageSize) {
+        this._staticHugePageSizes = _staticHugePageSizes;
+        this._staticDefaultHugePageSize = _staticDefaultHugePageSize;
+        this._thpMode = _thpMode;
+        this._thpPageSize = _thpPageSize;
+    }
+
+    @java.lang.Override
+    public String toString() {
+        return "Configuration{" +
+                "_staticHugePageSizes=" + _staticHugePageSizes +
+                ", _staticDefaultHugePageSize=" + _staticDefaultHugePageSize +
+                ", _thpMode=" + _thpMode +
+                ", _thpPageSize=" + _thpPageSize +
+                '}';
+    }
+
+    @java.lang.Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        HugePageConfiguration that = (HugePageConfiguration) o;
+        return _staticDefaultHugePageSize == that._staticDefaultHugePageSize && _thpPageSize == that._thpPageSize && Objects.equals(_staticHugePageSizes, that._staticHugePageSizes) && _thpMode == that._thpMode;
+    }
+
+    @java.lang.Override
+    public int hashCode() {
+        return Objects.hash(_staticHugePageSizes, _staticDefaultHugePageSize, _thpMode, _thpPageSize);
+    }
+
+    private static long readDefaultHugePageSizeFromOS() {
+        Pattern pat = Pattern.compile("Hugepagesize: *(\\d+) +kB");
+        Scanner scanner;
+        try {
+            scanner = new Scanner(new File("/proc/meminfo"));
```

Comment:
Nit: You can use try-with-resources here.

Reply:
Makes sense